pkg/batch/jobs/cacher/featureflagcacher.go

```diff
@@ -71,7 +72,11 @@ func (c *featureFlagCacher) Run(ctx context.Context) error {
 			c.logger.Error("Failed to list features", zap.String("environmentId", env.Id))
 			return err
 		}
-		if err := c.cache.Put(&ftproto.Features{Features: features}, env.Id); err != nil {
+		fts := &ftproto.Features{
+			Id:       evaluation.GenerateFeaturesID(features),
```

Comment:
Here, we generate an ID based on all the features per environment.
The same ID will be returned to the SDK so it can be checked on the server when the SDK requests the features from the cache.